ow_plexil/src/plexil-adapter/OwInterface.cpp

```diff
@@ -328,12 +333,32 @@ void OwInterface::managePanTilt (const string& opname,
 void OwInterface::cameraCallback (const sensor_msgs::Image::ConstPtr& msg)
 {
   // NOTE: the received image is ignored for now.
-
+  m_pointCloudRecieved = false;
   if (operationRunning (Op_TakePicture)) {
+    ros::Rate rate(10);
+    int timeout = 0;
+    // We wait for the pointcloud as well or the 5 sec timeout before marking as finished
+    while(m_pointCloudRecieved == false && timeout < 50){
+        ros::spinOnce();
+        rate.sleep();
+        timeout+=1;
+    }
+    if(timeout == 50){
+      ROS_ERROR("Timeout Exceeded: Recieved an Image but no PointCloud2.");
+    }
     mark_operation_finished (Op_TakePicture, Running.at (Op_TakePicture));
   }
 }
 
+void OwInterface::pointCloudCallback (const sensor_msgs::PointCloud2::ConstPtr& msg)
+{
+  // NOTE: the received pointcloud is ignored for now.
+  if (operationRunning (Op_TakePicture)) {
+    //mark as recieved
+    m_pointCloudRecieved = true;
+    mark_operation_finished (Op_TakePicture, Running.at (Op_TakePicture));
```

Comment:
I don't think we want to mark the operation finished here, since that is done in the cameraCallback, which waits for this callback.

Reply:
Good catch, I must have left that in accidentally when copy pasting!